### PR TITLE
Gradient support for API 1.8

### PR DESCRIPTION
**Pull request by EugeneElkin**

### CODE REVIEW

File: templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts [hunk 1]
```diff
@@ -0,0 +1,1381 @@
...
+        update<T>(options: VisualUpdateOptions, viewModel?: T): void;
+
+        /** Notifies the visual that it is being destroyed, and to do any cleanup necessary (such as unsubscribing event handlers). */
+        destroy?(): void;
+
+        /** Gets the set of objects that the visual is currently displaying. */
+        enumerateObjectInstances?(options: EnumerateVisualObjectInstancesOptions): VisualObjectInstanceEnumeration;
+    }
+
+    export interface IVisualHost extends extensibility.IVisualHost {
```
Comment (shaym83) [inline templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: This should be:

    export interface IVisualHost extends extensibility.IVisualHost {
        createSelectionIdBuilder: () => visuals.ISelectionIdBuilder;
        createSelectionManager: () => ISelectionManager;
        colorPalette: IColorPalette;
        persistProperties: (changes: VisualObjectInstancesToPersist) => void;
        applyJsonFilter: (filter: IFilter, objectName: string, propertyName: string, action: FilterAction) => void;
        tooltipService: ITooltipService;
        telemetry: ITelemetryService;
        locale: string;
        allowInteractions: boolean;
        instanceId: string;
        refreshHostData: () => void;
    }

File: templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts [hunk 2]
```diff
@@ -0,0 +1,1381 @@
...
+    const enum VisualEventType {
+        /** Should be used at the beginning of a visual's rendering operation. */
+        RenderStarted = 0,
+        /** Should be used at the end of a visual's rendering operation. */
+        RenderCompleted = 1,
+        /** Should be used by visuals to trace information in PBI telemetry. */
+        Trace = 2,
+        /** Should be used by visuals to trace errors in PBI telemetry. */
+        Error = 3,
+    }
```
Comment (shaym83) [inline templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: You should add this as well:

    const enum FilterAction {
        /** Merging filter into existing filters. */
        merge = 0,
        /** removing existing filter. */
        remove = 1,
    }

Comment (EugeneElkin) [reply templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: Thanks! Done!

File: templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts (hunk 1 above)
Comment (EugeneElkin) [reply templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: Thanks! Done!

Comment (shaym83) [reply templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: I don't see this change.
applyJosnFilter is outdated, instanceId, & refreshHostData are missing.
Did you publish a new iteration?

File: templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts (hunk 2 above)
Comment (shaym83) [reply templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: I also don't see this change anywhere.

File: templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts (hunk 1 above)
Comment (EugeneElkin) [reply templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: I created a new commit with manual changes today. All necesasry properties must be here.

File: templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts (hunk 2 above)
Comment (EugeneElkin) [reply templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: I created a new commit with manual changes today. FilterAction must be here. Please see line 84

File: templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts (hunk 1 above)
Comment (shaym83) [reply templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts]: Please remove launchUrl() - we are exposing this API only in 1.9

**Pull request merged**